bind.go

```diff
@@ -184,3 +183,24 @@ func In(query string, args ...interface{}) (string, []interface{}, error) {
 
 	return buf.String(), newArgs, nil
 }
+
+func appendReflectSlice(args []interface{}, v reflect.Value, vlen int) []interface{} {
+	switch val := v.Interface().(type) {
+	case []interface{}:
+		args = append(args, val...)
+	case []int:
+		for i := range val {
+			args = append(args, val[i])
```

Comment:
Currently each append requires a call to runtime.convT2E, which in turn allocates an interface{} per call. This means for a slice of len N we have N interface{} allocations.

If we append pointers instead this allocations can be avoided. This is safe as ints and strings are immutable. The only problem is that the slice itself is not immutable and the values inside the slice can change. We can solve this by making a copy of val and taking pointers from the copy.

This way we can replace N allocations from runtime.convT2E with 1 allocation from make and a call to copy.

Compared to this version I get these numbers:

```
name          old time/op    new time/op    delta
In-8             700ns ± 1%     629ns ± 1%  -10.10%  (p=0.000 n=10+10)
In1k-8          17.4µs ± 1%    18.1µs ± 1%   +3.93%  (p=0.000 n=10+9)
In1kInt-8       40.4µs ± 0%    21.5µs ± 0%  -46.76%  (p=0.000 n=9+10)
In1kString-8    56.5µs ± 1%    25.7µs ± 1%  -54.47%  (p=0.000 n=10+10)

name          old alloc/op   new alloc/op   delta
In-8              632B ± 0%      640B ± 0%   +1.27%  (p=0.000 n=10+10)
In1k-8          22.8kB ± 0%    22.8kB ± 0%     ~     (all equal)
In1kInt-8       30.8kB ± 0%    31.0kB ± 0%   +0.62%  (p=0.000 n=10+10)
In1kString-8    38.8kB ± 0%    39.2kB ± 0%   +0.99%  (p=0.000 n=10+10)

name          old allocs/op  new allocs/op  delta
In-8              12.0 ± 0%       8.0 ± 0%  -33.33%  (p=0.000 n=10+10)
In1k-8            6.00 ± 0%      6.00 ± 0%     ~     (all equal)
In1kInt-8        1.01k ± 0%     0.01k ± 0%  -99.30%  (p=0.000 n=10+10)
In1kString-8     1.01k ± 0%     0.01k ± 0%  -99.30%  (p=0.000 n=10+10)
```

And compared to the current performance on master:

```
name          old time/op    new time/op    delta
In-8             733ns ± 0%     629ns ± 1%  -14.21%  (p=0.000 n=9+10)
In1k-8          36.1µs ± 0%    18.1µs ± 1%  -49.79%  (p=0.000 n=10+9)
In1kInt-8       58.5µs ± 0%    21.5µs ± 0%  -63.22%  (p=0.000 n=10+10)
In1kString-8    73.4µs ± 1%    25.7µs ± 1%  -64.91%  (p=0.000 n=10+10)

name          old alloc/op   new alloc/op   delta
In-8              552B ± 0%      640B ± 0%  +15.94%  (p=0.000 n=10+10)
In1k-8          29.1kB ± 0%    22.8kB ± 0%  -21.52%  (p=0.000 n=10+10)
In1kInt-8       37.1kB ± 0%    31.0kB ± 0%  -16.36%  (p=0.000 n=10+10)
In1kString-8    45.1kB ± 0%    39.2kB ± 0%  -13.03%  (p=0.000 n=10+10)

name          old allocs/op  new allocs/op  delta
In-8              11.0 ± 0%       8.0 ± 0%  -27.27%  (p=0.000 n=10+10)
In1k-8            11.0 ± 0%       6.0 ± 0%  -45.45%  (p=0.000 n=10+10)
In1kInt-8        1.01k ± 0%     0.01k ± 0%  -99.31%  (p=0.000 n=10+10)
In1kString-8     1.01k ± 0%     0.01k ± 0%  -99.31%  (p=0.000 n=10+10)
```